### PR TITLE
WebGLRenderer: Make use of SRGB8_ALPHA8.

### LINKED ISSUE #22483: glTF conformance: khronos-TextureLinearInterpolationTest
I've finally updated our <model-viewer> fidelity tests and three.js is looking quite respectable (with the exception of sheen), but I did notice a clear failure: https://modelviewer.dev/fidelity/#khronos-TextureLinearInterpolationTest

This [warning](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/TextureLinearInterpolationTest) looks like it may be directed at us. I also notice many of our renders are subtly darker than the others if you look through these fidelity tests. Considering the exposures and tone mappings have been carefully matched, I wonder if this might be the cause?

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: Fixed #22483

**Description**

This PR introduces the usage of `SRGB8_ALPHA8`. That means sRGB encoded RGBA textures are now decoded via the WebGL API and not in the shader anymore. Now the decode into linear color space happens first and then the texture filtering is applied which produces more correct renderings.

The engine only uses `SRGB8_ALPHA8` with WebGL 2 and only with RGBA textures. This combination ensures mipmaps can be generated. Since RGBA is required and RGB textures are often emulated by WebGL and are in fact RGBA textures (https://github.com/mrdoob/three.js/pull/22293#discussion_r696763207), RGBA is now always used in various loaders.

It was necessary to update some screenshots since the visuals are a bit different than before in some cases. A good example is `webgl_loader_gltf_variants`.

### CODE REVIEW

**Comment by mrdoob** [comment]
Fascinating...

File: src/loaders/TextureLoader.js [hunk 1]
```diff
@@ -22,11 +21,6 @@ class TextureLoader extends Loader {
 		loader.load( url, function ( image ) {
 
 			texture.image = image;
-
-			// JPEGs can't have an alpha channel, so memory can be saved by storing them as RGB.
-			const isJPEG = url.search( /\.jpe?g($|\?)/i ) > 0 || url.search( /^data\:image\/jpeg/ ) === 0;
-
-			texture.format = isJPEG ? RGBFormat : RGBAFormat;
```
Comment (mrdoob) [inline src/loaders/TextureLoader.js:29]: @Oletus FYI

**Comment by mrdoob** [comment]
Is this expected?

image

/cc @WestLangley

**Comment by mrdoob** [comment]
This one seems strange too:

image

**Comment by Mugen87** [comment]
I had some problems with generating the E2E screenshots so I had to remove `--use-gl=swiftshader` from the config. It seems the existing setup does not work with the extensions. Without using swiftshader, some screenshots were not correctly produced. It seems I've overlooked  `webgl_materials_physical_transmission`.

The light probe example indeed renders differently.

**Comment by mrdoob** [comment]
Uh oh, force-pushing messes with the Pull Request conversation.. 😶

**Comment by Mugen87** [comment]
Sorry, I push additional commits from now on 👍 .

**Comment by mrdoob** [comment]
> I had some problems with generating the E2E screenshots so I had to remove `--use-gl=swiftshader` from the config. It seems the existing setup does not work with the extensions.

Hmm, which extensions?

**Comment by Mugen87** [comment]
I mean `SRGB8_ALPHA8 `. The E2E screenshots showed black textures in the example which would rely on `SRGB8_ALPHA8`.

**Comment by mrdoob** [comment]
Oh, so SwiftShader doesn't support `SRGB8_ALPHA8`?

**Comment by Mugen87** [comment]
At least I could not get it to work on my iMac. But it seems the GitHub backend has no problems with it.

**Comment by mrdoob** [comment]
This one is also weird:

image

Seems like the horses `receiveShadow` is not working.

**Comment by mrdoob** [comment]
> At least I could not get it to work on my iMac. But it seems the GitHub backend has no problems with it.

Have you tried doing `npm update` in the `/test` folder?

**Comment by Mugen87** [comment]
I'll give it a try.

**Comment by mrdoob** [comment]
The black edges are finally gone on this one! 🎉

image

**Comment by Mugen87** [comment]
It seems it does not solve the issue. `webgl_animation_skinning_blending` looks like so now (although it renders correctly in the browser).

webgl_animation_skinning_blending

**Comment by mrdoob** [comment]
@Mugen87 Oh well. Thanks for testing 🙏

**Comment by mrdoob** [comment]
> The black edges are finally gone on this one! 🎉
>
> image

Although I wonder if this is related to this PR or if it was a previous change and the E2E didn't catch it.

@Mugen87 Could you include the builds in the PR so we can compare raw.githack links with current dev?

**Comment by Mugen87** [comment]
Sure!

Dev: https://raw.githack.com/mrdoob/three.js/447e88a55c47fd9163360c6be957b784d098f352/examples/index.html
Prod: https://threejs.org/examples/

**Comment by mrdoob** [comment]
Another thing that would help is...
Instead of updating all the screenshots, can you update only the screenshots that fail in the E2E?

**Comment by mrdoob** [comment]
Edit: I'm saying this because I don't see a different between these two:

https://raw.githack.com/Mugen87/three.js/dev17/examples/index.html?q=car#webgl_materials_car
https://raw.githack.com/mrdoob/three.js/dev/examples/index.html?q=car#webgl_materials_car

But the screenshots are different.

**Comment by Mugen87** [comment]
It seems the SSR examples look wrong now. However, they do something which is not expected in the post-processing demos.

https://github.com/mrdoob/three.js/blob/c789b20ee6ba1cf41fad7ba52e1564fc4bf78cb6/examples/webgl_postprocessing_ssr.html#L147

When using post-processing, sRGB should be normally applied via `GammaCorrectionShader` at the end of the pass chain. It seems `SSRPass` and `SSRrPass` have an encoding logic that is not consistent regarding this policy.

**Comment by Mugen87** [comment]
Okay, I've just updated the screenshots that failed. Although I'm not understand the logic why some test fail and others not. E.g. `webgl_loader_gltf_variants` passes although the is a noticeable hue shift.

https://raw.githack.com/Mugen87/three.js/dev17/examples/webgl_loader_gltf_variants.html
https://threejs.org/examples/webgl_loader_gltf_variants

**Comment by Mugen87** [comment]
> When using post-processing, sRGB should be normally applied via GammaCorrectionShader at the end of the pass chain. It seems SSRPass and SSRrPass have an encoding logic that is not consistent regarding this policy.

I've updated the PR with the respective change.

**Comment by mrdoob** [comment]
> Although I'm not understand the logic why some test fail and others not. E.g. webgl_loader_gltf_variants passes although the is a noticeable hue shift.

Yes, that is its own issue. Better not to go into that rabbit hole right now. I've been there 😬

**Comment by Mugen87** [comment]
> The light probe example indeed renders differently.

That was indeed a bug in `PMREMGenerator`. Its shaders also perform inline decodes/encodes which are not necessary when working with `SRGB8_ALPHA8`.

**Comment by mrdoob** [comment]
> Okay, I've just updated the screenshots that failed. Although I'm not understand the logic why some test fail and others not. E.g. `webgl_loader_gltf_variants` passes although the is a noticeable hue shift.
>
> https://raw.githack.com/Mugen87/three.js/dev17/examples/webgl_loader_gltf_variants.html
> https://threejs.org/examples/webgl_loader_gltf_variants

I think that is unrelated to this PR. I can see the hue shift already happening in `dev`:

https://raw.githack.com/mrdoob/three.js/dev/examples/webgl_loader_gltf_variants.html

My guess is this was due the HDR clamping fix: https://github.com/mrdoob/three.js/pull/22451

**Comment by mrdoob** [comment]
Okay, I've gone through the examples and everything looks good!👌

Can you remove the builds so this can be merged?

**Comment by Mugen87** [comment]
> Can you remove the builds so this can be merged?

I have reverted the build commits 👍

**Comment by mrdoob** [comment]
Thanks!

File: src/loaders/TextureLoader.js (hunk 1 above)
Comment (Oletus) [reply src/loaders/TextureLoader.js:29]: This seems like an acceptable compromise to ensure generateMipmap works, devices tend to have a lot more memory these days compared to when this change was made.

**Pull request merged**